### ISSUE TITLE
Upgrade to node-gyp 10.x

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-theia/theia#13670]
### Feature Description:
Right now, we need to install the old python3 version 3.10 in order for our build to work due to https://github.com/nodejs/node-gyp/issues/2869. (removal of `distutils`). 

Updgrading to node-gyp 10+ should fix this problem.

**Comment by msujew** [comment]
FYI, while I agree that we should update to node-gyp 10, newer versions of python3 are supported once you install `setuptools` manually into your python installation (via `pip install setuptools`).

**Comment by tsmaeder** [comment]
If that is the alternative, we need to update our build processes & documentation. Currently we install python 3.10. Also: installing setuptools does not seem to work for everyone, if I read the internet correctly.

**Comment by buddha314** [comment]
I am having the issue with `distutils` and `setuptools` is not solving the problem. FWIW building on Windows 11 or WSL is a tremendous pain in the butt.

**Comment by tsmaeder** [comment]
@buddha314 I am building on Windows 11 daily. As long as I use the proper Python version, I don't have any problems. Could you be more detailed about what is not working for you?

**Comment by buddha314** [comment]
> [@buddha314](https://github.com/buddha314) I am building on Windows 11 daily. As long as I use the proper Python version, I don't have any problems. Could you be more detailed about what is not working for you?

Probably worth it's own thread. But Python is the most annoying. I don't see an official installer for 3.11 or 3.10 anymore https://www.python.org/downloads/windows/  Also, the "py" shorthand breaks multiple other things.  DO you have notes you could post as a guide? I'd send you good vibes!

**Comment by bsgdigital** [comment]
The documentation clearly needs an update - currently it says "Python3 is required for the build due to node-gyp@8.4.1"

I'm using 3.12 and it didn't work without setuptools install.

I've also briefly tried gitpod and there was a similar problem (though may be a bit different, haven't spent much time with it) - in any case, it failed.

**Comment by tsmaeder** [comment]
> I'm using 3.12 and it didn't work without setuptools install.

Yes, the instructions for Windows state max is 3.11. Once we get rid of node 18 support, we'll be able to update node-gyp to a newer version.

**Comment by tsmaeder** [comment]
@ndoschek
